### PR TITLE
Excel: Enable filling of selected building block with terms with fill button

### LINKED ISSUE #536: [Feature Request]: Swate, Excel, restore Functionality of button "Fill selected cells with this term"
### Contact Details

_No response_

### Is your feature request related to a problem?

Currently the button responses with the Message: "not implemented"

### Describe the solution you would like!

The button shall fill the selected cells with the selected term and corresponding information

### Additional context!

_No response_

### Host

Excel

### Host Details

Excel

### PR DESCRIPTION
**Pull request by Etschbeijer**
closes #536

### CODE REVIEW

File: src/Client/OfficeInterop/OfficeInterop.fs [hunk 1]
```diff
@@ -181,6 +181,9 @@ module OfficeInteropExtensions =
                         column.getRange().columnHidden <- shallHide)
             }
 
+        static member parseToObj (input: string) : obj =
```
Comment (Freymaurer) [inline src/Client/OfficeInterop/OfficeInterop.fs:184]: remove this and just use box

File: src/Client/OfficeInterop/OfficeInterop.fs [hunk 2]
```diff
@@ -2665,6 +2664,91 @@ let updateTopLevelMetadata (arcFiles: ArcFiles) =
...
+                    let firstIndex = Array.head columnIndices
+                    let lastIndex = Array.last columnIndices
+
+                    let isUnit = Array.contains columnHeaders.[2] tableHeaders.[firstIndex..lastIndex] //Unit
+                    for rowIndex in firstRow..(firstRow + rowCount-1.) do
+                        
+                        columnIndices
+                        |> Array.iter (fun columnIndex ->
+                            match tableHeaders.[columnIndex] with
+                            | header when header = columnHeaders.[2] -> //Unit
```
Comment (Freymaurer) [inline src/Client/OfficeInterop/OfficeInterop.fs:2722]: This match case looks familiar, was this used somewhere else?

**Review by Freymaurer — CHANGES_REQUESTED** [review]

File: src/Client/OfficeInterop/OfficeInterop.fs (hunk 2 above)
Comment (Etschbeijer) [reply src/Client/OfficeInterop/OfficeInterop.fs:2722]: Yes, a similiar matching was used for the term rectify

File: src/Client/OfficeInterop/OfficeInterop.fs (hunk 1 above)
Comment (Etschbeijer) [reply src/Client/OfficeInterop/OfficeInterop.fs:184]: fixed

File: src/Client/OfficeInterop/OfficeInterop.fs [hunk 3]
```diff
@@ -2665,6 +2664,93 @@ let updateTopLevelMetadata (arcFiles: ArcFiles) =
...
+                            |> Option.defaultValue ""
+                            |> (fun s -> s.TrimEnd())
+                        )
+                    )
+
+                let tableHeaders = tableValues.[0]
+                let firstRow = selectedRange.rowIndex
+                let rowCount = selectedRange.rowCount
+
+                if firstRow = 0 && rowCount = 1 then return [InteropLogging.Msg.create InteropLogging.Error "You cannot fill the headers of an annotation table with terms!"]
```
Comment (Freymaurer) [inline src/Client/OfficeInterop/OfficeInterop.fs:2701]: Maybe do:

1. Unspecialized insert if not in annotation table
2. If in annotation table do:
    1. Get selected building block
    2. Check building block type
    3. Insert accordingly

Comment (Etschbeijer) [reply src/Client/OfficeInterop/OfficeInterop.fs]: Should be fixed

**Pull request merged**